horace_core/GUI/horace.m

```diff
@@ -2109,15 +2121,21 @@ function Rebin_rebin_pushbutton_Callback(hObject, eventdata, handles)
         save(out,outfilename);
     end
 
-catch
-    the_err=lasterror;
-    emess=the_err.message;
-    nchar=strfind(emess,['at ',num2str(the_err.stack(1).line)]);
-    mess1='No rebin performed';
-    mess2=emess(nchar+9:end);
-    set(handles.message_info_text,'String',char({mess_initialise,mess1,mess2}));
+catch the_err
+    err = {sprintf('**** %s',mess_initialise),...
+        sprintf('**** Error %s Message: %s',...
+        the_err.identifier,the_err.message),...
+        '**** rebin failed at: '};
+    stack_depth = numel(the_err.stack);
+    for i=stack_depth-3:-1:1
+        err{end+1} = sprintf('*** Row: %d, function: %20s file: %s',...
+            the_err.stack(i).line,the_err.stack(i).name,the_err.stack(i).file);
+    end
+    
+    set(handles.message_text_field,'String',char(err));
     guidata(gcbo,handles);
-    return;
+    
+    rethrow(the_err);
```

Comment:
Maybe all these error catch / printing to the GUI could be hived off into a separate function which is called rather than repeating the code >10 times?

Reply:
had such though. May be if its two, it worth implementing. Doing it only for master